controllers/clusters/helpers.go

```diff
@@ -202,10 +204,42 @@ func createSpecDifferenceMessage(k8sSpec, iSpec any) (string, error) {
 		return "", err
 	}
 
-	msg := "There are external changes on the Instaclustr console. Please reconcile the specification manually. "
-	specDifference := fmt.Sprintf("k8s spec: %s; data from instaclustr: %s", k8sData, iData)
+	var k8sSpecMap map[string]any
+	err = json.Unmarshal(k8sData, &k8sSpecMap)
+	if err != nil {
+		return "", err
+	}
+
+	var iSpecMap map[string]any
+	err = json.Unmarshal(iData, &iSpecMap)
+	if err != nil {
+		return "", err
+	}
+
+	const externalChangesBaseMsg = "There are external changes on the Instaclustr console. Please reconcile the specification manually."
+
+	diffs := dcomparison.MapsDiff("spec", k8sSpecMap, iSpecMap)
+	if len(diffs) == 1 {
+		return fmt.Sprintf(
+			"%s Diff: {path: %s, k8sValue: %s, instaclustrValue: %s}",
+			externalChangesBaseMsg,
+			diffs[0].Path,
+			diffs[0].Value1,
+			diffs[0].Value2,
+		), nil
+	}
+
+	var diffMessages []string
+	for _, diff := range diffs {
+		diffMessages = append(diffMessages, fmt.Sprintf(
+			"{field: %s, k8sValue: %v, instaclustrValue: %v}",
+			diff.Path,
+			diff.Value1,
+			diff.Value2,
+		))
+	}
 
-	return msg + specDifference, nil
+	return fmt.Sprintf("%s Diffs: %s", externalChangesBaseMsg, strings.Join(diffMessages, ", ")), nil
```

Comment:
It will looks a bit strange if use `instaclustrValue` and `k8sValue` inside methods of `ObjectDiff`. I decided to create a func that prepares the diff message.